engine/api/src/main/scala/pl/touk/nussknacker/engine/api/dict/static/StaticDictRegistry.scala

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.api.dict.static
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.engine.api.dict.{DictDefinition, DictRegistry}
+import pl.touk.nussknacker.engine.api.dict.DictRegistry.{DictEntryWithKeyNotExists, DictEntryWithLabelNotExists, DictNotDeclared}
+
+trait StaticDictRegistry extends DictRegistry {
+
+  protected def declarations: Map[String, DictDefinition]
+
+  override def keyByLabel(dictId: String, label: String): Validated[DictRegistry.DictLookupError, String] = {
+    declarations.get(dictId).map(Valid(_)).getOrElse(Invalid(DictNotDeclared(dictId))).andThen {
+      case static: StaticDictDefinition =>
```

Comment:
yes, I kind of understand this, I more meant:
        static.keyByLabel.get(label).map(key => Valid(key))
          .getOrElse(Invalid(DictEntryWithLabelNotExists(dictId, label, Some(static.keyByLabel.keys.toList))))
==> why this is not part of static?

Reply:
It can be part of `StaticDictDefinition` but then user's implementation which will check the same but in some database/service will do it in `DictRegistry` or pass fat, managed object like connection pool as a method's parameter to his own `DictDefinition`. For first option we will have inconsistency (part of logic in `DictRegistry`, part of logic in  `StaticDictDefinition`. For the second one we will pass managed object as a parameter that is a bad design. Or maybe I still misunderstand your point?